Mathlib/AlgebraicGeometry/EllipticCurve/Weierstrass.lean

```diff
@@ -535,7 +534,29 @@ structure EllipticCurve (R : Type u) [CommRing R] extends WeierstrassCurve R whe
 
 namespace EllipticCurve
 
-variable {R : Type u} [CommRing R] (E : EllipticCurve R)
+variable {R : Type u} [CommRing R]
+
+section ext
+
+theorem ext' : Function.Injective (toWeierstrassCurve : EllipticCurve R → _)
+  | ⟨x1, _, x3⟩, ⟨y1, _, y3⟩, h => by
+    change x1 = y1 at h
+    congr
+    exact Units.ext (by rw [x3, y3, h])
+
+variable {x y : EllipticCurve R}
+
+theorem ext_iff' : x = y ↔ x.toWeierstrassCurve = y.toWeierstrassCurve :=
+  ⟨fun h ↦ by rw [h], fun h ↦ EllipticCurve.ext' h⟩
```

Comment:
Then this is `toWeierstrassCurve_injective.eq_iff.symm`, if you need it. (Do you?)

Reply:
Good point! Actually I don't need it, though.